src/audio/src/src_generic.c

```diff
@@ -66,9 +66,11 @@ static inline void fir_filter_generic(int32_t *rp, const void *cp, int32_t *wp0,
 			data++;
 			coef++;
 		}
-		if (data == fir_end)
-			data = fir_start;
 
+		/* No need to check for circular wrap. Pointer data is moved to
+		 * fir_start to be used by next loop if n2 is greater than zero.
+		 */
+		data = fir_start;
```

Comment:
not sure the reason for a FIR with 2 loops, please help clarify
also, for simply, change to below:
		for (i = 0; i < n1; i++) {
			y0 += (int64_t)(*coef) * (*data++);
			y1 += (int64_t)(*coef++) * (*data++);
		}
no risk for change like this.

Reply:
There's need to do circular wrap for pointer data. It happens after calculate of n1 filter taps. After wrap calculate remaining n2 taps.

Something like this should work but I find it messy looking. I don't know if it saves any MCPS but I can try.

`y1 += (int64_t)(*(coef++)) * (*(data++));`

Reply:
The above change works OK. It doesn't seem to improve MCPS but since the code is more compact I'll do it.

Reply:
ok, so our plan is to use intrinsic in src_hifi3.c for circular wrap, right?